File: go/is_leap_year.go
package main

import "fmt"

var years = []int{1900, 2000, 2023, 2024, 2400}

func isLeap(year int) bool {
	return year%4 == 0 && year%100 != 0 || year%400 == 0
}

func main() {
	for _, year := range years {
		fmt.Printf("is %d a leap year? %t \n", year, isLeap(year))
	}
}

File: JavaScript/isLeapYear.js
function isLeapYearWithoutLib(year, usLib) {
    return ((year % 4 === 0) && (year % 100 !== 0)) || (year % 400 === 0);
}

function isLeapYearWithLib(year) {
    return new Date(year, 1, 29).getDate() === 29;
}

const dates = [1900, 2000, 2023, 2024, 2400];

for (const year of dates) {
    console.log(`is ${year} a leap year? ${isLeapYearWithLib(year)}`);
    console.log(`is ${year} a leap year without lib? ${isLeapYearWithoutLib(year)}`);
}
